utils/funcexp/func_timediff.cpp

```diff
@@ -162,16 +162,35 @@ string Func_timediff::getStrVal(rowgroup::Row& row,
         case execplan::CalpontSystemCatalog::CHAR:
         case execplan::CalpontSystemCatalog::TEXT:
             text = parm[0]->data()->getStrVal(row, isNull);
-            isTime1 = std::count(text.begin(), text.end(), '-') <= 1; // Time can have at most 1 dash (signifies negative)
-            val1 = dataconvert::DataConvert::stringToDatetime(text, &isDate1);
+            
+            if (text.length() >= 14) // datetime has length at least 14
```

Comment:
This is the major difference between results for MDB and columnstore. We should be converting all non-temporal data types to times if length is less than 14. There was a special case i found while testing results. see comment on line 170

Reply:
I wonder if this magic value 14 differ for some of locales? Did you investigate this case?

Reply:
BTW did you consider the case when the table uses UTF-8 encoding for varchar/text columns? This must produce widechar string but it worths to check b/c magic values tend to change in some corner cases.